packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -256,6 +257,98 @@ class ArtifactBuilder {
+  /// with the system installation of Flutter to reset
+  /// `.dart_tool/package_config.json` after the build completes or fails.
+  Future<IpaBuildResult> buildMacos({
+    bool codesign = true,
+    String? flavor,
+    String? target,
+    List<String> args = const [],
+    String? base64PublicKey,
+    DetailProgress? buildProgress,
+  }) async {
+    final projectRoot = shorebirdEnv.getShorebirdProjectRoot()!;
+    // Delete the .dart_tool directory to ensure that the app is rebuilt.
+    // Without this, the build command will not print the app.dill path
+    final dartToolDir = Directory(p.join(projectRoot.path, '.dart_tool'));
+    if (dartToolDir.existsSync()) {
+      dartToolDir.deleteSync(recursive: true);
+    }
+
+    String? appDillPath;
+    await _runShorebirdBuildCommand(() async {
+      const executable = 'flutter';
+      final arguments = [
+        'build',
+        'macos',
+        '--release',
+        if (flavor != null) '--flavor=$flavor',
+        if (target != null) '--target=$target',
+        if (!codesign) '--no-codesign',
+        ...args,
+      ];
+
+      final buildProcess = await process.start(
+        executable,
+        arguments,
+        runInShell: true,
+        environment: base64PublicKey?.toPublicKeyEnv(),
+      );
+
+      final stdoutLines = <String>[];
+      buildProcess.stdout
+          .transform(utf8.decoder)
+          .transform(const LineSplitter())
+          .listen((line) {
+        stdoutLines.add(line);
+        if (buildProgress == null) {
+          return;
+        }
+
+        final update = _progressUpdateFromIpaBuildLog(line);
+        if (update != null) {
+          buildProgress.updateDetailMessage(update);
+        }
+      });
+
+      final stderrLines = await buildProcess.stderr
+          .transform(utf8.decoder)
+          .transform(const LineSplitter())
+          .toList();
+      final stderr = stderrLines.join('\n');
+      final stdout = stdoutLines.join('\n');
+      final exitCode = await buildProcess.exitCode;
+
+      // If we've been updating the progress, reset it to the original base
+      // message so as not to leave the user with a confusing message.
+      buildProgress?.updateDetailMessage(null);
+
+      if (exitCode != ExitCode.success.code) {
+        throw ArtifactBuildException('Failed to build: $stderr');
+      }
+
+      if (stderr.contains('Encountered error while creating the IPA')) {
+        final errorMessage = _failedToCreateIpaErrorMessage(stderr: stderr);
+
+        throw ArtifactBuildException('''
+Failed to build:
+$errorMessage''');
+      }
+
+      appDillPath = findAppDill(stdout: stdout);
+    });
+
+    if (appDillPath == null) {
+      throw ArtifactBuildException('''
+Unable to find app.dill file.
+Please file a bug at https://github.com/shorebirdtech/shorebird/issues/new with the logs for this command.
+''');
+    }
+
+    return IpaBuildResult(kernelFile: File(appDillPath!));
```

Comment:
Shouldn't we make a separate result type rather than sharing the IpaBuildResult for iOS?

Reply:
Yeah, I think we'll probably want to unify a bunch of the linking behavior that exists in all of the apple targets. Not sure what we call it.

Reply:
Called it MacosBuildResult. We'll fix it later.